modules/RegeneronGeneBurden.py

```diff
@@ -0,0 +1,326 @@
+
+MARKER_TO_METHOD_DESC = {
+    "M1.singleton": "Burden test carried out with singleton pLOF variants.",
+    "M1.0001": "Burden test carried out with pLOF variants with a MAF smaller than 0.001%.",
+    "M1.001": "Burden test carried out with pLOF variants with a MAF smaller than 0.01%.",
+    "M1.01": "Burden test carried out with pLOF variants with a MAF smaller than 0.1%.",
+    "M1.1": "Burden test carried out with pLOF variants with a MAF smaller than 1%.",
+    "M3.singleton": "Burden test carried out with singleton pLOF and deleterious missense variants.",
+    "M3.0001": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.001%.",
+    "M3.001": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.01%.",
+    "M3.01": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.1%.",
+    "M3.1": "Burden test carried out with pLOFs and deleterious missense variants with a MAF smaller than 1%.",
+}
+
+
+def main(regeneron_data: str, gwas_studies: str, spark_instance: SparkSession) -> DataFrame:
+    """
+    This module extracts and processes target/disease evidence from the raw data published in PMID:34662886.
+    Args:
+    """
+    logging.info(f"File with the results from the REGENERON burden analyses: {regeneron_data}")
+    logging.info(f"File with GWAS Catalog studies: {gwas_studies}")
+
+    # Load data
+
+    TO_KEEP = [
+        "Gene",
+        "Trait",
+        "Trait type",
+        "UKB detailed trait name",
+        "Trait_String",
+        "Marker",
+        "Marker type",
+        "P-value",
+        "Effect (95% CI)",
+        "N cases with 0|1|2 copies of effect allele",
+        "N controls with 0|1|2 copies of effect allele",
+        "Ancestry",
+    ]
+
+    eur_evd_df = (
+        spark_instance.createDataFrame(
+            read_excel(regeneron_data, sheet_name="SD2").filter(items=TO_KEEP).replace({nan: None}).astype(str)
+        )
+        .withColumn("Ancestry", lit("EUR"))
+        .withColumnRenamed("UKB detailed trait name", "Study tag")
+    )
+
+    non_eur_evd_df = spark_instance.createDataFrame(
+        read_excel(regeneron_data, sheet_name="SD3").filter(items=TO_KEEP).replace({nan: None}).astype(str)
+    ).withColumnRenamed("Trait_String", "Study tag")
+
+    summary_stats_df = spark_instance.createDataFrame(
+        read_excel(regeneron_data, sheet_name="SD4", skiprows=0, header=1)
+        .iloc[2:]
+        .reset_index(drop=True)
+        .filter(items=["Study tag", "Study Accession"])
+        .replace({nan: None})
+        .astype(str)
+    ).withColumn("Study tag", split("Study tag", "__SV").getItem(0))
+
+    gwas_studies_df = (
+        spark_instance.read.csv(gwas_studies, header=True, inferSchema=True, sep="\t")
+        .filter(col("LINK").contains("34662886"))
```

Comment:
I know it is fine here, but i'm wondering why not to use the PMID column? In an edge case the above solution can be inaccurate.  
```suggestion
        .filter(col('PUBMEDID') == '34662886')
```

Reply:
Fixed.